### PR TITLE
Marine

### PR DESCRIPTION
**Pull request by nathanemac**

### CODE REVIEW

File: src/Meta/marine_population_dynamics.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+marine_population_dynamics_meta = Dict(
+    :nvar => 352,  
+    :variable_nvar => false,  
+    :ncon => 330,  
+    :variable_ncon => false,  
+    :minimize => true, 
+    :name => "marine_population_dynamics",
+    :has_equalities_only => false,
+    :has_inequalities_only => true,  
+    :has_bounds => " ⋯ 461 bytes ⋯ " ult_nvar, kwargs...) = 0
```
Comment (dpo) [inline src/Meta/marine_population_dynamics.jl:10]: What's this?

**Review by dpo — CHANGES_REQUESTED** [review]

File: src/ADNLPProblems/marine_population_dynamics.jl [hunk 2]
```diff
@@ -0,0 +1,98 @@
+using ADNLPModels, NLPModelsIpopt, Plots, Statistics
+
+function marine_population_dynamics(;n::Int = default_nvar, tf=10000, ϵ = 1e-4, type::Val{T} = Val(Float64), kwargs...) where {T}
+
+    n > 21 && @warn("arwhead: number of timesteps must be lower than 21. Solving the problem for n=21:")
```
Comment (tmigot) [inline src/ADNLPProblems/marine_population_dynamics.jl:5]: Replace `arwhead` by `marine`

File: src/ADNLPProblems/marine_population_dynamics.jl [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    cut_data = data[1:n,1:ns]
+    data_r=reshape(cut_data,ns*n,1)
+
+    # definition of the objective function
+    function objective(y::AbstractVector{T}) where T
+        s=0
+        for i=1:ns*n
+            s+=(y[i+2ns]-data_r[i])^2
+        end
+        return s
```
Comment (tmigot) [inline src/ADNLPProblems/marine_population_dynamics.jl:44]:
```
function f(x)
  return sum(....... for i=1:ns*n)
end
```

File: src/ADNLPProblems/marine_population_dynamics.jl [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    # definition of constraint function
+    function c(y::AbstractVector{T}) where T
+
+        c_1 = y[1:ns] # growth rates < 0
+        c_2 = y[1:ns] .+ y[ns+1:2ns] # growth rates + mortality rates < 0
+
+        c_3 = similar(y, n * ns -1)
+        for i = 2:ns*n
+            c_3[i-1] = y[2ns+i] - y[2ns+i-1] - y[2ns+ns*n+i-1]*dt #constraint of Euler
+        end
```
Comment (tmigot) [inline src/ADNLPProblems/marine_population_dynamics.jl:56]: c3 = [....... for i=1:(ns*n-1)]

File: src/ADNLPProblems/marine_population_dynamics.jl [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        c_4 = similar(y, n*(ns-1))
+        for k=2:ns
+            for j=1:n
+                c_4[(k-2)n+j] = y[(ns+n*ns) + (k-1)*n+j] - y[k-1]*y[2ns+(k-2)n+j] + (y[4]+y[ns+k])*y[2ns + (k-1)n+j]
+            end
```
Comment (tmigot) [inline src/ADNLPProblems/marine_population_dynamics.jl:62]: Same comment as for c3

File: src/ADNLPProblems/marine_population_dynamics.jl [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        C=[c_1;
+        c_2;
+        c_3;
+        c_4]
+
+        return C
+    end
+
+    # Declaring the bounds on the constraints
+    lcon_c1 = -ones(ns) .- ϵ
```
Comment (tmigot) [inline src/ADNLPProblems/marine_population_dynamics.jl:74]: Use `ones(T, ns)`, instead of `ones(ns)` and same comment for `zeros` and `rand` below.

**Review by tmigot — CHANGES_REQUESTED** [review]
Thanks @nathanemac ! I have a couple of comments.

**Comment by dpo** [comment]
@tmigot What is the status here?

**Comment by tmigot** [comment]
I believe the model here is a simplified version of the one described in `marine.mod` in the archive provided in https://github.com/JuliaSmoothOptimizers/OptimizationProblems.jl/issues/161 .

**Pull request closed without merging**